components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java

```diff
@@ -0,0 +1,269 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.oidc;
+
+import org.apache.axis2.client.Stub;
+import org.apache.oltu.oauth2.client.response.OAuthClientResponse;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.IObjectFactory;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.ObjectFactory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.common.model.ClaimMapping;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@PrepareForTest({LoggerFactory.class, URL.class})
+public class OpenIDConnectAuthenticatorTest {
+
+/*
+    @Mock
+    private URL mockUrl;*/
+    @Mock
+    private HttpServletRequest mockRequest;
+    @Mock
+    private OAuthClientResponse mockResponse;
+    @Mock
+    private OpenIDConnectAuthenticator mockAuthenticator;
+    @Mock
+    private AuthenticationContext mockAuthenticationContext;
+    @Mock
+    private HttpURLConnection mockConnection;
+    @InjectMocks
+    private OpenIDConnectAuthenticator openIDConnectAuthenticator;
+
+    private static AuthenticationContext context = null;
+    private static Map<String, String> authenticatorProperties;
+    private static String accessToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9";
+    private static OAuthClientResponse token;
+    private static Map<String, Object> jsonObject;
+
+    @BeforeTest
+    public void init() {
+        //MockitoAnnotations.initMocks(this);
+        openIDConnectAuthenticator = new OpenIDConnectAuthenticator();
+        authenticatorProperties = new HashMap<>();
+        authenticatorProperties.put("callbackUrl","http://localhost:8080/playground2/oauth2client");
+        authenticatorProperties.put("commonAuthQueryParams", "Show error equals true.");
+        authenticatorProperties.put("UserInfoUrl","https://localhost:9443/oauth2/userinfo");
+
+        jsonObject = new HashMap<>();
+        Stub stub = new Stub() {
+            @Override
+            public int hashCode() {
+                return super.hashCode();
+            }
+        };
+        jsonObject.put("stub", stub);
+        token = null;
+
+    }
+
+    @Test
+    public void testCanHandle() throws Exception {
+
+        mockRequest = mock(HttpServletRequest.class);
```

Comment:
why not move this to three separate test cases with some comments explaining what each test cases focuses on